### DISCUSSION TITLE
Equivalent Circuit Model with various conditions

### DISCUSSION THREAD

**Discussion by hokonkim** [pybamm-team/PyBaMM#2791] [Q&A]
Hello, PyBaMM communities

I start to use PyBaMM for my work.

Now I'm trying to model equivalent circuit model (ECM) from the scratch.

I would like to model some various conditions.
(e.g., 1 resistor + 0 RC (simplest)
        1 resistor + 1 RC (simplest), etc..)

Based on the document related to Thevenin model, I changed the number of RC from 1 (default) to 0.

But the result from 0 RC looks the same with that from 1 RC.

If I would like to start from 1 resistor + 0 RC condition first, then what am I supposed to do?

Here is the code that I used.

    import pybamm
    pybamm.set_logging_level("INFO")
    options = {'operating mode': 'current', 'number of rc elements': 0}
    model = pybamm.equivalent_circuit.Thevenin(name='Thevenin Equivalent Circuit Model', options=options)
    experiment = pybamm.Experiment(
    [
        (
            "Discharge at C/10 for 10 hours or until 3.3 V at 25oC",
            "Rest for 30 minutes at 25oC",
            "Charge at 10 A until 4.1 V at 25oC",
            "Rest for 30 minutes at 25oC",
        ),
    ] * 6
    )
    sim = pybamm.Simulation(model, experiment=experiment)
    sim.solve()
    sim.plot(colors='blue')

 

In addition, when trying RC elements more than 1 (e.g., RC elements=2), then I got the error message like this:
"'R2 [Ohm]' not found. Best matches are ['R1 [Ohm]', 'R0 [Ohm]']"

I would like to see how the change will happens depending on the element of ECM.

**Comment by balaji3112** [comment]
I guess you need to define the new terms(R2) in the parameter file where R1, R0 were already defined.
